test/endpoints/favourite_filters/test_save_filter.py

```diff
@@ -0,0 +1,32 @@
+from bson import ObjectId
+from fastapi.testclient import TestClient
+
+from test.endpoints.favourite_filters.conftest import FavouriteFilter
+
+
+class TestSaveFilter:
+    def test_save_filter(
+        self,
+        test_app: TestClient,
+        login_and_get_token,
+    ):
+        name = "Test Create Filter"
+        filter_content = "PM-201-HJ-CRY-FLOW > 10"
+
+        test_response = test_app.post(
+            f"/users/filters?name={name}&filter={filter_content}",
+            headers={"Authorization": f"Bearer {login_and_get_token}"},
+        )
+
+        assert test_response.status_code == 201
+
+        # Testing the response is a valid object ID
+        filter_id = test_response.json()
+        ObjectId(filter_id)
+
+        test_filter = FavouriteFilter()
+        result = test_filter.find_one(filter_id)
+        print(f"Result 123: {result}")
+        assert result.get("filters") is not None
```

Comment:
In light of the point above about defaulting to empty strings, I think it would be better here if we asserted on the contents, so we can be sure that the behaviour is desirable. E.G. rather than just asserting `is not None`:
```suggestion
    assert len(result.get("filters")) == 1
    assert "_id" in result.get("filters")[0]
    assert result.get("filters")[0]["name"] == "Test Create Filter"
    assert result.get("filters")[0]["filter"] == "PM-201-HJ-CRY-FLOW > 10"
```